ext/native-decls/RegisterCommand.md

```diff
@@ -19,24 +19,31 @@ The restricted bool is not used on the client side. Permissions can only be chec
 ![](https://i.imgur.com/TaCnG09.png)
 
 
-## Parameters
+## Required Arguments
 * **commandName**: The command you want to register.
 * **handler**: A handler function that gets called whenever the command is executed.
+
+## Optional Arguments
 * **restricted**: If this is a server command and you set this to true, then players will need the command.yourCommandName ace permission to execute this command.
 
+## Handler function parameters
+* **playerSource**: The player font which executed the command.
+* **args**: returns the string the player entered after the command.
+* **rawCommand**: returns the string with the name of the executed command.
+
 ## Examples
 
 ```lua
 -- (server side script)
 -- Registers a command named 'ping'.
-RegisterCommand("ping", function(source, args, rawCommand)
-    -- If the source is > 0, then that means it must be a player.
-    if (source > 0) then
+RegisterCommand("ping", function(playerSource, args, rawCommand)
```

Comment:
Doesn't make much sense to change this for the sake of change.

Reply:
Excuse me because it was using the logic that I talked about using 'source' as the first determined parameter, this breaks a rule in the original documentation of LUA.

Reply:
> Excuse me because it was using the logic that I talked about using 'source' as the first determined parameter, this breaks a rule in the original documentation of LUA.

I'm not even sure what you mean by 'the original documentation of Lua', 'source' is referred to as the sender **everywhere** in documentation, there doesn't need to be a place where it deviates.

Reply:
`The server ID, canonically known as the NetID, is a transient handle that is usually seen in server scripts as source.` from the [network](https://docs.fivem.net/docs/scripting-manual/networking/ids/) documentation.

Registered in other places as source [#1](https://docs.fivem.net/docs/scripting-manual/migrating-from-deprecated/creating-commands/) [#2](https://docs.fivem.net/docs/scripting-manual/introduction/creating-your-first-script-csharp/) [#3](https://docs.fivem.net/docs/scripting-manual/introduction/creating-your-first-script-javascript/) [#4](https://docs.fivem.net/docs/scripting-manual/introduction/creating-your-first-script/)

Reply:
@hiroshi-nagaza This change breaks a convention used across literally every resource in the FiveM ecosystem. FiveM doesn't need to abide by weird, archaic Lua rules; especially when everyone familiar with the project knows what this variable means.

Reply:
For example, the event that checks when an element touches a marker, was for the event to return: (The element which entered the marker)

in this case, the source would be the marker, and it would not be a parameter defined in the function, understand?

Reply:
> @Hiroshi-Nagaza This change breaks a convention used across literally every resource in the FiveM ecosystem. FiveM doesn't need to abide by weird, archaic Lua rules; especially when everyone familiar with the project knows what this variable means.

oh yes, excuse me then, it is because I am very young and didn’t know you used 'source' for everything.

Reply:
but thank you very much, but I would like that no documentation of each function would hear a list of arguments to which it returns in a function.

Reply:
> oh yes, excuse me then, it is because I am very young and didn’t know you used 'source' for everything.

Then why are you trying to change the documentation? 🤔

Reply:
hello

Reply:
> > oh yes, excuse me then, it is because I am very young and didn’t know you used 'source' for everything.
> 
> Then why are you trying to change the documentation?

It is because I am used to the MTA: SA platform, a platform that many members from there are updating and coming here, after all you offer an experience in GTA V and they in GTA SA

Reply:
But in the case of 'source' is it variable? if i use it in a script can i use it for example: `thePlayer`?

Reply:
> But in the case of 'source' is it variable? if i use it in a script can i use it for example: `thePlayer`?

'source' is the person who called the command